src/validation.cpp

```diff
@@ -941,9 +941,12 @@ bool ReadBlockFromDisk(CBlock& block, const FlatFilePos& pos, const Consensus::P
         return error("%s: Deserialize or I/O error - %s at %s", __func__, e.what(), pos.ToString());
     }
 
-    // Check the header
-    if (!CheckProofOfWork(block.GetHash(), block.nBits, consensusParams))
-        return error("ReadBlockFromDisk: Errors in block header at %s", pos.ToString());
+    // We skip POW checks for genesis block
+    if (block.GetHash() != consensusParams.hashGenesisBlock) {
```

Comment:
I would move this test inside `CheckProofOfWork`

Reply:
I need this if block for `CheckBlockSolution` later, so I would have to do the if case anyway.